app/ts/store/tokens.ts

```diff
@@ -0,0 +1,112 @@
+import { signal, useComputed, useSignalEffect } from '@preact/signals'
+import { useAccount } from './account.js'
+
+const CACHEID_PREFIX = '_ut'
+
+const retrieveTokensFromCache = (cacheKey: string) => {
+	const tokensCache = localStorage.getItem(cacheKey)
+
+	if (tokensCache === null) {
+		return DEFAULT_TOKENS
+	}
+
+	try {
+		return JSON.parse(tokensCache) as TokenMeta[]
+	} catch {
+		throw new Error('Tokens cache may be corrupted.')
+	}
+}
+
+const tokens = signal<TokenMeta[]>([])
+
+export function useAccountTokens() {
+	const { address } = useAccount()
+
+	const cacheKey = useComputed(() => {
+		if (address.value.state !== 'resolved') return `${CACHEID_PREFIX}:default`
+		return `${CACHEID_PREFIX}:${address.value.value}`
+	})
+
+	const addToken = (newToken: TokenMeta) => {
+		tokens.value = [...tokens.value, newToken]
+	}
+
+	const listenForCacheKeyChange = () => {
+		tokens.value = retrieveTokensFromCache(cacheKey.value)
+	}
+
+	const listenForTokensChange = () => {
+		if (cacheKey.value === `${CACHEID_PREFIX}:default`) return
+		const newCache = JSON.stringify(tokens.value)
+		localStorage.setItem(cacheKey.value, newCache)
+	}
+
+	useSignalEffect(listenForCacheKeyChange)
+	useSignalEffect(listenForTokensChange)
+
+	return {
+		tokens,
+		addToken,
+	}
+}
+
+// Move to constants later
+export type TokenMeta = {
+	chainId: number
+	name: string
+	address: string
```

Comment:
I think it would be better if these are bigints when parsed from storage and then stored as strings to storage

Reply:
this change may break so i created #102 to address later